app/src/main/java/swati4star/createpdf/activity/MainActivity.java

```diff
@@ -512,20 +514,35 @@ public void setNavigationViewSelection(int id) {
         mNavigationView.setCheckedItem(id);
     }
 
-    private void getRuntimePermissions() {
-        PermissionsUtils.checkRuntimePermissions(this,
-                PERMISSION_REQUEST_CODE,
-                Manifest.permission.WRITE_EXTERNAL_STORAGE,
-                Manifest.permission.READ_EXTERNAL_STORAGE,
-                Manifest.permission.CAMERA);
+    private boolean getRuntimePermissions() {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            if ((ContextCompat.checkSelfPermission(this, Manifest.permission.WRITE_EXTERNAL_STORAGE)
+                    != PackageManager.PERMISSION_GRANTED) ||
+                    (ContextCompat.checkSelfPermission(this, Manifest.permission.CAMERA)
+                            != PackageManager.PERMISSION_GRANTED) ||
+                    (ContextCompat.checkSelfPermission(this, Manifest.permission.READ_EXTERNAL_STORAGE)
+                            != PackageManager.PERMISSION_GRANTED)) {
+                requestPermissions(new String[]{
+                        Manifest.permission.WRITE_EXTERNAL_STORAGE,
+                        Manifest.permission.READ_EXTERNAL_STORAGE,
+                        Manifest.permission.CAMERA},
+                        0);
+                return false;
+            }
+        }
+        return true;
     }
 
     private boolean isStoragePermissionGranted() {
-        boolean permission = PermissionsUtils.checkRuntimePermissions(this,
-                PERMISSION_REQUEST_CODE,
-                Manifest.permission.WRITE_EXTERNAL_STORAGE,
-                Manifest.permission.READ_EXTERNAL_STORAGE);
-        return permission;
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            if ((ContextCompat.checkSelfPermission(this, Manifest.permission.WRITE_EXTERNAL_STORAGE)
+                    != PackageManager.PERMISSION_GRANTED) &&
+                    (ContextCompat.checkSelfPermission(this, Manifest.permission.READ_EXTERNAL_STORAGE)
+                            != PackageManager.PERMISSION_GRANTED)) {
+                return false;
+            }
+        }
+        return true;
```

Comment:
Why have you changed permissions code?

Reply:
i am still thinking when i changed this :thinking: 
when i have merged with swati master channel it was automatically added to my code 

lemme change it